--- picamorig/streams.py	2019-02-09 10:31:10.594232000 +0000
+++ picamchanges/streams.py	2019-02-09 10:38:40.442655000 +0000
@@ -518,7 +518,12 @@
                     # no need to adjust self._pos_offset
                 else:
                     # need to remove the head of the chunk
-                    self._data[0] = chunk[self._length - self._size:]
+                    try:
+                        self._data[0] = chunk[self._length - self._size:]
+                    except TypeError:
+                        self._length=round(self._length)
+                        self._size=round(self._size)
+                        self._data[0] = chunk[self._length - self._size:]
                     self._pos -= self._length - self._size
                     self._length = self._size
             return result
